Classes/Controller/AbstractController.php

```diff
@@ -162,4 +165,35 @@ protected function resolveView()
 	{
 		return null;
 	}
+
+
+	/**
+	 * Injects the Configuration Manager and is initializing the framework settings
+	 *
+	 * @param \TYPO3\CMS\Extbase\Configuration\ConfigurationManagerInterface $configurationManager Instance of the Configuration Manager
+	 */
+	public function injectConfigurationManager( ConfigurationManagerInterface $configurationManager )
+	{
+		// set ConfigurationManager and get settings like the FrontendConfigurationManager
+		$this->configurationManager = $configurationManager;
+		$this->settingsOrg = $this->configurationManager->getConfiguration( ConfigurationManagerInterface::CONFIGURATION_TYPE_SETTINGS );
+
+		$this->settings = $this->settingsOrg;
+
+		// load settings from template setup
+		$setup = $this->configurationManager->getConfiguration( ConfigurationManagerInterface::CONFIGURATION_TYPE_FULL_TYPOSCRIPT );
+		$extensionName = 'aimeos';
+
+		// get plugin settings and override with typo3 default settings
+		if( is_array( $setup['plugin.']['tx_' . strtolower( $extensionName ) . '.'] ) )
+		{
+			$configuration = Base::convertTypoScriptArrayToPlainArray( $setup['plugin.']['tx_' . strtolower( $extensionName ) . '.'] );
+
+			if( is_array( $configuration['settings'] ) )
+			{
+				$this->settings = \TYPO3\CMS\Extbase\Utility\ArrayUtility::arrayMergeRecursiveOverrule( $configuration['settings'], $this->settingsOrg, false, false );
```

Comment:
if the settings are found override it with the values loaded from t3 (pages, plugins, etc) but as before only override values which are set to something, no null values